_rules/iframe-non-empty-title-attribute-cae760.md

```diff
@@ -0,0 +1,208 @@
+---
+id: cae760
+name: '`iframe` element has non-empty title attribute'
+rule_type: atomic
+description: |
+  This rule checks that each `iframe` element has a non-empty `title` attribute.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+    - Wilco Fiers
+---
+
+## Applicability
+
+The rule applies to `iframe` elements, except if the `iframe` is not part of [sequential focus navigation][], and one of the following is true:
+
+- the `iframe` has a [hidden state][] of `true`; or
+- the `iframe` has an [explicit role][] of `presentation` or `none`; or
+- the `iframe` has no [content document][] with [content][] [included in the accessibility tree][].
+
+## Expectation
+
+Each target element has a `title` [attribute value][] that is neither empty (`""`) nor only [ASCII whitespace][].
+
+## Assumptions
+
+This rule assumes that an `iframe` is only a [user interface component][] if it is a container that assistive technologies can navigate into, or if it is a stop in the [sequential focus navigation][]. Since [4.1.2 Name Role Value][sc412] only applies to [user interface components][user interface component] not all `iframe` elements require a `title` attribute, including some `iframe` elements that are announced by assistive technologies.
+
+## Accessibility Support
+
+There are substantial differences between how various browsers and assistive technologies treat `iframe` elements. This rule is designed to work in a variety of commonly used combinations of them. Some notable differences are:
+
+- Some browsers do not use the [accessible name computation][] to compute the name of an `iframe` element, and instead only announce the `title` attribute.
```

Comment:
I put it here because this is neither an assumption about the content, nor an assumption about how to interpret the requirement. The "choice" that is made is to not exclude voiceover+safari from this rule, which I think the rule properly documents. I think this needs to be here.

Reply:
The problem is that the examples with `aria-label` are now false positive for JAWS users.

We're choosing to follow one UA+AT and not the others, which we have to do, but we're making the choice to follow the one that goes directly against the specs: https://www.w3.org/TR/html-aam-1.0/#iframe-element which doesn't sound good.

I'm not sure this solves #1367 and would like to see @dd8 opinion on this.

Reply:
Not a single one of them respects the specs JY. We have three options here:

1. Follow the spec, ignoring the fact that not a single browser actually works the way it is spec'ed. This was the first version, and the ACT TF we've had pushback on that, including from AG and the ACT TF. So that option is out.

2. Write a rule to the lowest common denominator; if it works anywhere at all, pass it. Which as far as I can see, means iframes never fail because Chrome+JAWS just doesn't treat iframes as UICs. We would just have to deprecate the rule. That's doens't seem like a particularly viable option either.

3. We intentionally leave out some of the major browsers, even though none of them work according to spec. That seems like picking favorites, which doens't seem fair.

4. Write a rule that, ignoring spec, makes sure that iframes are accessible in all major browsers and screen readers. That is this version. If this rule says its a fail, it is definitely a fail somewhere. Not in all of them mind you, but definitely somewhere. This is in line with "accessibility support" as WCAG defines it, and seems to be what Mark was suggesting we do too.

If we want an iframe accessible name rule at all, I think this is the only viable option. It's a little controversial. But it's the least controversial option I could figure out. Open to other ideas, but I'm not on board with picking favorites or ditching the rule.

Reply:
A solution I would be much more confortable with:
* Keep "each target has a non-empty accessible name" as expectation. At this point we don't really care where the accessible name comes from.
* Remove examples that use `aria-label`, `aria-labelledby` or anything other than `title`. These examples behave differently on different UA/AT, making them fairly annoying for us whether they pass or fail, since we don't want to take sides.
* Add an Accessibility Support in the line of "some UA+AT only use `title` for accessible name of `iframe`, while other also use `aria-label` and `aria-labelledby`, depending on UA+AT, the same target may pass or fail." + maybe strengthen with "it is recommended to use `title` only since this will be correctly handled by all UA+ATs"
* + likely copy the AS note in the def of accessible name, since it is a discrepancy about accessible name computation and behaviours that differ from the specs.